src/libsyntax/attr.rs

```diff
@@ -357,6 +464,26 @@ pub fn last_meta_item_value_str_by_name(items: &[P<MetaItem>], name: &str)
 
 /* Higher-level applications */
 
+pub fn sort_meta_item_list_vals(values: Vec<MetaListItem>) -> Vec<MetaListItem> {
+    values.into_iter().map(|Spanned {node, span}| Spanned {
+        node: match node {
+            MetaListItemKind::MetaItem(i) => MetaListItemKind::MetaItem(sort_meta_item(i)),
+            _ => node
+        },
+        span: span
+    }).collect()
+}
+
+pub fn sort_meta_item(item: P<MetaItem>) -> P<MetaItem> {
+    item.map(|Spanned {node, span}| Spanned {
+        node: match node {
+            MetaItemKind::List(n, vs) => MetaItemKind::List(n, sort_meta_item_list_vals(vs)),
+            _ => node
+        },
+        span: span
+    })
+}
```

Comment:
Both of these functions are just the identity (neither calls `sort_meta_items`), which does the real work.

n.b. `sort_meta_items` appears to be unused (both by `rustc` and by any plugins on Github) so it and these two functions you added can probably be removed.